packages/configraph/src/entity.ts

```diff
@@ -126,57 +77,226 @@ export class ConfigraphEntity<
   injectedPlugins: Array<ConfigraphPlugin> = [];
   ownedPlugins: Array<ConfigraphPlugin> = [];
 
-  state: any;
-  private overrideSources = [] as Array<OverrideSource>;
-
   // /** error encountered while _loading_ the config */
   // readonly configLoadError?: ConfigLoadError;
 
   /** error within the schema itself */
   readonly schemaErrors: Array<SchemaError> = [];
 
+  // private templateChain: Array<ConfigraphEntityTemplate<EntityMetadata, NodeMetadata>> = [];
+  public defs: Array<ConfigraphEntityDef<EntityMetadata, NodeMetadata>> = [];
+
+  readonly graphRoot: Configraph;
+  readonly templateRootEntity?: ConfigraphEntity<EntityMetadata, NodeMetadata>;
+
   constructor(
```

Comment:
This is where the significant changes are.

we now follow up a chain of templates, applying the changes to config nodes from each, and adding/deleting/removing additional entities to the graph if applicable